src/Crypto/Signing/HsmSignerService.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 De Staat der Nederlanden, Ministerie van Volksgezondheid, Welzijn en Sport.
+// Licensed under the EUROPEAN UNION PUBLIC LICENCE v. 1.2
+// SPDX-License-Identifier: EUPL-1.2
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using NL.Rijksoverheid.ExposureNotification.BackEnd.Domain;
+
+namespace NL.Rijksoverheid.ExposureNotification.BackEnd.Crypto.Signing
+{
+    /// <summary>
+    /// Typed HttpClient to request a signature from the HsmSignerService API
+    /// </summary>
+    public class HsmSignerService : IHsmSignerService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly IHsmSignerConfig _config;
+
+        public HsmSignerService(
+            HttpClient httpClient,
+            IHsmSignerConfig config
+            )
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+
+            _httpClient.BaseAddress = new Uri(_config.BaseAddress);
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        public async Task<byte[]> GetNlSignatureAsync(byte[] content)
+        {
+            using var sha256Hasher = SHA256.Create();
+            var contentHash = sha256Hasher.ComputeHash(content);
+            var contentHashBased64 = Convert.ToBase64String(contentHash);
+
+            var certificateChain = GetCertificateChain();
```

Comment:
I think we can get rid of this and just store the entire chain in nlPem here...If that's part of the clean up, then just ignore me 😄

Reply:
Indeed you are correct... removed the offending code 🥳